### DISCUSSION TITLE
Outside temperature

### DISCUSSION THREAD

**Discussion by JoeDet** [Hyundai-Kia-Connect/kia_uvo#577] [Ideas]
Hi.

Im wanted to use historical data of kW/km and outside temperature to improve range calculation for my Ioniq 5 EV, is there a way to add outside temp measurements?

**Comment by cdnninja** [comment]
Can you check if this data is in your data sensor's attributes?  If so we can add it.

**Reply by JoeDet** [reply]
I have a "sensor.ioniq_5_set_temperature", but that looks like its the internal AC temp reading.

In the "sensor.ioniq_5_data" I have som temps, but none of them looks like it outside temp and formatting is unknown to me. Closest thing I can find is under vehicleStatus (but also looks like hvac):

`  airTemp:
    value: 02H
    unit: 0
    hvacTempType: 1`

**Reply by cdnninja** [reply]
That value is your set temp.  We convert it to a real number.  Since it isn't in the response it's probably best to create something in home assistant looking up weather using the gps location of the car.

**Comment by 6zXo8aMn2TRXOs** [comment]
There's an example of creating a Home assistant rest sensor here https://community.home-assistant.io/t/get-current-temperature-for-my-cars-location/396051
